Libs/MRML/Core/vtkMRMLSegmentationStorageNode.cxx

```diff
@@ -862,35 +862,42 @@ int vtkMRMLSegmentationStorageNode::ReadBinaryLabelmapRepresentation(vtkMRMLSegm
         {
           currentBinaryLabelmap = vtkSmartPointer<vtkOrientedImageData>::New();
 
-          // Extent
-          int currentSegmentExtent[6] = { 0, -1, 0, -1, 0, -1 };
-          std::string currentExtentString;
-          if (this->GetSegmentMetaDataFromDicitionary(currentExtentString, dictionary, segmentIndex, KEY_SEGMENT_EXTENT))
+          // Get layer extent: union of all segment extents in this layer
+          int layerExtent[6] = { 0, -1, 0, -1, 0, -1 };
+          for (int segmentIndexForExtent : segmentIndexInLayer[frameIndex])
           {
-            GetImageExtentFromString(currentSegmentExtent, currentExtentString);
-          }
-          else
-          {
-            vtkWarningToMessageCollectionMacro(this->GetUserMessages(), "vtkMRMLSegmentationStorageNode::ReadBinaryLabelmapRepresentation",
-              "Segment extent is missing for segment " << segmentIndex);
-            for (int i = 0; i < 6; i++)
+            int currentSegmentExtent[6] = { 0, -1, 0, -1, 0, -1 };
+            std::string currentExtentString;
+            if (this->GetSegmentMetaDataFromDicitionary(currentExtentString, dictionary, segmentIndexForExtent, KEY_SEGMENT_EXTENT))
             {
-              currentSegmentExtent[i] = imageExtentInFile[i];
+              GetImageExtentFromString(currentSegmentExtent, currentExtentString);
```

Comment:
If the extent string is invalid, then the extent of the current segment will be skipped.
Should we instead switch to the imageExtentInFile if we encounter an invalid extent string?

Reply:
By invalid do you mean empty (something like [0, -1, 0, -1, 0, -1])? That would mean that the segment is empty and so it should not expand the extent.

Reply:
No, I meant that if the string itself was invalid, and could not be correctly parsed.

Reply:
Indeed all these parsing codes are still not fully robust for invalid input. The changes in this commit improve things, so I would integrate it as is, and implement further improvements later if needed.